torch/csrc/jit/codegen/cuda/scheduler/normalization.cpp

```diff
@@ -645,17 +659,48 @@ TORCH_CUDA_CU_API c10::optional<ReductionParams> getNormalizationHeuristics(
       n_tensor_inputs > 0,
       "Tried to schedule a fusion with no tensor inputs, currently not supported.");
 
-  auto persistent_buffers = scheduler_utils::persistentBuffers(fusion);
+  HeuristicCacheAccessor<scheduler_utils::PersistentBufferInfo>
+      persistent_buffer_data;
+
+  // TODO: move all these boilerplate code into the accessor class
+  // (follow up)
+  if (data_cache && !data_cache->isRecording()) {
+    persistent_buffer_data.writeTemporary(
+        data_cache->getPersistentBufferInfo());
+  } else {
+    persistent_buffer_data.writeNew(scheduler_utils::persistentBuffers(fusion));
+    if (data_cache && data_cache->isRecording()) {
+      data_cache->setPersistentBufferInfo(persistent_buffer_data.read());
+    }
+  }
+
+  auto& persistent_buffers = persistent_buffer_data.read();
   bool requires_persistence = !persistent_buffers.buffers.empty();
 
   auto properties =
       scheduler_utils::getProperties(fusion, runtime_info, first_red_tv);
 
-  auto max_persistent_size =
-      scheduler_utils::persistentBufferSize(fusion, runtime_info);
+  auto max_persistent_size = scheduler_utils::persistentBufferSize(
+      fusion, runtime_info, persistent_buffers, data_cache);
+
+  HeuristicCacheAccessor<std::vector<TensorView*>>
+      vectorizable_inputs_outputs_data;
```

Comment:
I would expect this to also be in reduction.cpp, but don't see it.

Reply:
I see. I overlooked caching for that part. It was re-computed at runtime for reduction scheduler. I will add caching for this one.